framework/midi/internal/fluidlitesynth.cpp

```diff
@@ -171,42 +171,31 @@ Ret FluidLiteSynth::setupChannels(const Programs& programs)
         return false;
     }
 
-    m_programs = programs;
-
     fluid_synth_program_reset(m_fluid->synth);
     fluid_synth_system_reset(m_fluid->synth);
 
-    for (const Program& prog : m_programs) {
-        fluid_synth_reset_tuning(m_fluid->synth, prog.channel);
-
-        fluid_synth_bank_select(m_fluid->synth, prog.channel, prog.bank);
-        fluid_synth_program_change(m_fluid->synth, prog.channel, prog.program);
+    std::set<channel_t> channels;
+    for (const Event& e: events) {
+        channels.insert(e.channel);
+    }
 
-        fluid_synth_set_interp_method(m_fluid->synth, prog.channel, FLUID_INTERP_DEFAULT);
-        fluid_synth_pitch_wheel_sens(m_fluid->synth, prog.channel, 12);
+    for (channel_t ch : channels) {
+        fluid_synth_reset_tuning(m_fluid->synth, ch);
+        fluid_synth_set_interp_method(m_fluid->synth, ch, FLUID_INTERP_DEFAULT);
+        fluid_synth_pitch_wheel_sens(m_fluid->synth, ch, 12);
+    }
 
-        //LOGD() << "ch: " << prog.ch << ", prog: " << prog.prog << ", bank: " << prog.bank << "\n";
+    for (const Event& e: events) {
+        handleEvent(e);
     }
 
     return make_ret(Ret::Code::Ok);
```

Comment:
make_ret return an instance of Ret, but the function has return type bool

Reply:
fixed